src/policy_artifacthub.rs

```diff
@@ -0,0 +1,263 @@
+use anyhow::{anyhow, Result};
+use csv::ReaderBuilder;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use time::OffsetDateTime;
+use url::Url;
+
+use crate::policy_metadata::{self, MetadataV2};
+
+/// Format of artifacthub-pkg.yml file as defined in
+/// https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ArtifactHubPkg {
+    pub version: String, // TODO semver
+    pub name: String,
+    pub display_name: String,
+    #[serde(with = "time::serde::rfc3339")]
+    pub created_at: OffsetDateTime,
+    pub description: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub license: Option<String>, // TODO spdx
+    #[serde(rename = "homeURL", skip_serializing_if = "Option::is_none")]
+    pub home_url: Option<Url>, // https://github.com/kubewarden/verify-image-signatures
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub containers_images: Option<Vec<ContainerImage>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keywords: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub links: Option<Vec<Link>>,
+
+    // #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
I'm undecided on if to use `metadata.yml`'s `"io.kubewarden.policy.author"` to construct one or both `artifacthubpkg.yml`'s `provider` and `maintainers`. Hardcoding all `provider: kubewarden` may be ok, but having no distinction between policies owned by the Kubewarden team and 3rd parties is weird.

Reply:
Right now `io.kubewarden.policy.author` is a string that can hold multiple mail addresses. I would use that as input data to build the `maintainers` key, which is the only one that 1) allows multiple people to be used 2) has the concept of name/email.

I would just ignore `provider` and even leave it out of our struct

Reply:
Built maintainers out of `io.kubewarden.policy.author`.
Hardcoded `provider` to `kubewarden` as per RFC.